core/src/test/java/fr/sncf/osrd/DriverBehaviourTest.java

```diff
@@ -0,0 +1,68 @@
+package fr.sncf.osrd;
+
+import fr.sncf.osrd.envelope_sim_infra.MRSP;
+import fr.sncf.osrd.infra.InfraHelpers;
+import fr.sncf.osrd.infra.api.Direction;
+import fr.sncf.osrd.infra.implementation.tracks.directed.DirectedInfraBuilder;
+import fr.sncf.osrd.infra.implementation.tracks.directed.TrackRangeView;
+import fr.sncf.osrd.railjson.schema.common.graph.ApplicableDirection;
+import fr.sncf.osrd.railjson.schema.infra.trackranges.RJSApplicableDirectionsTrackRange;
+import fr.sncf.osrd.railjson.schema.infra.trackranges.RJSSpeedSection;
+import fr.sncf.osrd.reporting.warnings.DiagnosticRecorderImpl;
+import fr.sncf.osrd.train.TestTrains;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import java.util.HashMap;
+import java.util.List;
+
+public class DriverBehaviourTest {
+    @Test
+    public void mrspWithDriverBehaviour() {
+        List<TrackRangeView> path = pathSenarioForTest();
+        var testRollingStock = TestTrains.VERY_SHORT_FAST_TRAIN;
+        var driverBehaviour = new DriverBehaviour(16, 17);
+        var mrsp = MRSP.from(path, testRollingStock, true, null);
+        mrsp = driverBehaviour.applyToMRSP(mrsp);
+        Assertions.assertEquals(42, mrsp.interpolateSpeedRightDir(0, 1));
```

Comment:
This would be easier to follow if the math behind those values were explicit.

Something like:

    Assertions.assertEquals(42, mrsp.interpolateSpeedRightDir(firstSpeedTransition - brakingOffset, 1));

(you can name more variables for this)

You can also call `interpolateSpeedLeftDir` to check that we move from one speed to another at precisely the given position. We should in fact check every transition with both calls.